src/protocols/aeternity/helpers/actions.ts

```diff
@@ -0,0 +1,124 @@
+import { Encoded, Tag } from '@aeternity/aepp-sdk';
+import { ITransaction } from '@/types';
+import JsonBig from '@/lib/json-big';
+import { fetchJson } from '@/utils';
+import { useAeSdk } from '@/composables/aeSdk';
+import { useAeNetworkSettings } from '@/protocols/aeternity/composables';
+import { AeSdkSuperhero } from '@/protocols/aeternity/libs/AeSdkSuperhero';
+import { useStore } from 'vuex';
+import { useMiddleware } from '@/composables/middleware';
+
+async function fetchTransactionsFromMiddleware(
+  address: Encoded.AccountAddress,
+  nextPageUrl: string | null,
+  limit: number,
+) {
+  const {
+    fetchFromMiddlewareCamelCased,
+  } = useMiddleware();
+
+  const url = ([null, ''].includes(nextPageUrl))
+    ? `/v2/accounts/${address}/activities?limit=${limit}`
+    : nextPageUrl!;
+
+  try {
+    const x = await fetchFromMiddlewareCamelCased(url);
+
+    return {
+      regularTransactions: x.data || [],
+      nextPageParams: x.next,
+    };
+  } catch (error) {
+    return {};
+  }
+}
+
+export async function fetchPendingTransactions(
+  address: Encoded.AccountAddress,
+  sdk: AeSdkSuperhero,
+) {
+  try {
+    const fetchedPendingTransaction = (
+      await sdk.api.getPendingAccountTransactionsByPubkey(address)
+    );
+
+    return JsonBig.parse(JsonBig.stringify(
+      fetchedPendingTransaction?.transactions || [],
+    )) as ITransaction[];
+  } catch (error) {
+    return [];
+  }
+}
+
+async function fetchTipWithdrawnTransactions(address: Encoded.AccountAddress, recent: boolean) {
+  try {
+    const store = useStore();
+    const { getAeSdk } = useAeSdk({ store });
+    const { aeActiveNetworkSettings } = useAeNetworkSettings();
+    await getAeSdk();
+
+    const response = await fetchJson(
+      `${aeActiveNetworkSettings.value.backendUrl}/cache/events/?address=${address}&event=TipWithdrawn${recent ? '&limit=5' : ''}`,
+    );
+    if (response.message) {
+      return [];
+    }
```

Comment:
Isn't it a mistake?

Reply:
no, this is correct if (message -> error)